Tests/DatadogTests/Datadog/DatadogCore/DatadogCoreTests.swift

```diff
@@ -187,4 +188,50 @@ class DatadogCoreTests: XCTestCase {
         )
         XCTAssertEqual(requestBuilderSpy.requestParameters.count, 3, "It should send 3 requests")
     }
+
+    func testWhenPerformancePresetOverrideIsNotProvided() throws {
+        let core = DatadogCore(
+            directory: temporaryCoreDirectory,
+            dateProvider: RelativeDateProvider(advancingBySeconds: 0.01),
+            initialConsent: .granted,
+            userInfoProvider: .mockAny(),
+            performance: .mockRandom(),
+            httpClient: .mockAny(),
+            encryption: nil,
+            contextProvider: .mockAny(),
+            applicationVersion: .mockAny()
+        )
+        try core.register(
+            feature: FeatureMock(
+                name: "mock",
+                performanceOverride: nil
+            )
+        )
+        let feature = core.v2Features.values.first
+        XCTAssertEqual(feature?.storage.authorizedFilesOrchestrator.performance.maxObjectSize, UInt64(512).KB)
+        XCTAssertEqual(feature?.storage.authorizedFilesOrchestrator.performance.maxFileSize, UInt64(4).MB)
```

Comment:
nit/ These numbers (`512 KB` + `4 MB`) look magical in this context and they are defined [quite far](https://github.com/DataDog/dd-sdk-ios/blob/5a657d8ed96ff49834c08fda3b673932eb80df92/Sources/Datadog/Core/PerformancePreset.swift#L116-L126) from this test implementation. We already test default performance values [here](https://github.com/DataDog/dd-sdk-ios/blob/5a657d8ed96ff49834c08fda3b673932eb80df92/Tests/DatadogTests/Datadog/Core/PerformancePresetTests.swift#L12), so repeating these assertions doesn't bring much convenience nor clarity.

I believe that what we really want to assert is: _"When performance preset is overrided, then new values are used"_. This could be asserted in a single test that registers two Features, one with override and another one without, and asserting that the latter uses expected values in their storage (which are different for first Feature). That could be simpler (one test) and more scalable (no config repetition), WDYT @maciejburda ?

Reply:
I initially had it in one test, but I felt like it could use test name to improve clarity.

I think you're right. Let's put it back to one test - it shows the scope of the performance override better.